src/Common/ParagraphFormatHelper.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+
+namespace Microsoft.PowerShell.PlatyPS
+{
+    public class ParagraphFormatSettings
+    {
+        /// <summary>
+        /// The maximum length of a line in the paragraph.
+        /// This includes indentation.
+        /// </summary>
+        public ushort LineLength { get; set; }
+        
+        /// <summary>
+        /// How much indentation to add to each line.
+        /// </summary>
+        public byte Indentation { get; set; }
+
+        /// <summary>
+        /// how much indentation to add to the first line.
+        /// </summary>
+        public byte FirstLineIndentation { get; set; }
+
+        /// <summary>
+        /// The default settings for a paragraph arg:
+        /// LineLength = 80
+        /// No indentation
+        /// No first line indentation
+        /// </summary>
+        public ParagraphFormatSettings()
+        {
+            LineLength = 80;
+            Indentation = 0;
+            FirstLineIndentation = 0;
+        }
+
+        public ParagraphFormatSettings(ushort lineLength, byte indentation, byte firstLineIndentation)
+        {
+            LineLength = lineLength;
+            Indentation = indentation;
+            FirstLineIndentation = firstLineIndentation;
+        }   
+    }
+
+    /// <summary>
+    /// Helper class to format a string into a paragraph.
+    /// It can also indent the paragraph, and add extra indentation to the first line.
+    /// </summary>
+    public class ParagraphFormatHelper
+    {
+        public static string FormatParagraph(string input, ParagraphFormatSettings settings)
+        {
+            StringBuilder outputLine = new StringBuilder(); // used for formatting the entire paragraph
```

Comment:
Borrow a string build from the StringbuilderPool - https://github.com/PowerShell/platyPS/blob/v2/src/StringBuilderPool.cs